src/code/z_actor.c

```diff
@@ -294,34 +294,36 @@ void Target_InitReticle(TargetContext* targetCtx, s32 actorCategory, PlayState*
 void Target_SetNaviState(TargetContext* targetCtx, Actor* actor, s32 actorCategory, PlayState* play) {
     TargetColor* targetColor = &sTargetColorList[actorCategory];
 
-    targetCtx->naviRefPos.x = actor->focus.pos.x;
-    targetCtx->naviRefPos.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
-    targetCtx->naviRefPos.z = actor->focus.pos.z;
+    targetCtx->naviHoverPos.x = actor->focus.pos.x;
+    targetCtx->naviHoverPos.y = actor->focus.pos.y + (actor->targetArrowOffset * actor->scale.y);
+    targetCtx->naviHoverPos.z = actor->focus.pos.z;
 
-    targetCtx->naviInner.r = targetColor->inner.r;
-    targetCtx->naviInner.g = targetColor->inner.g;
-    targetCtx->naviInner.b = targetColor->inner.b;
-    targetCtx->naviInner.a = targetColor->inner.a;
+    targetCtx->naviInnerColor.r = targetColor->inner.r;
+    targetCtx->naviInnerColor.g = targetColor->inner.g;
+    targetCtx->naviInnerColor.b = targetColor->inner.b;
+    targetCtx->naviInnerColor.a = targetColor->inner.a;
 
-    targetCtx->naviOuter.r = targetColor->outer.r;
-    targetCtx->naviOuter.g = targetColor->outer.g;
-    targetCtx->naviOuter.b = targetColor->outer.b;
-    targetCtx->naviOuter.a = targetColor->outer.a;
+    targetCtx->naviOuterColor.r = targetColor->outer.r;
+    targetCtx->naviOuterColor.g = targetColor->outer.g;
+    targetCtx->naviOuterColor.b = targetColor->outer.b;
+    targetCtx->naviOuterColor.a = targetColor->outer.a;
 }
 
 void Target_Init(TargetContext* targetCtx, Actor* actor, PlayState* play) {
-    targetCtx->arrowPointedActor = targetCtx->lockOnActor = targetCtx->unk_8C = targetCtx->bgmEnemy = NULL;
+    targetCtx->naviHoverActor = targetCtx->reticleActor = targetCtx->forcedLockOnActor = targetCtx->bgmEnemy = NULL;
 
     targetCtx->reticleSpinCounter = 0;
     targetCtx->curReticle = 0;
-    targetCtx->unk_40 = 0.0f;
+    targetCtx->naviMoveProgressFactor = 0.0f;
 
     Target_SetNaviState(targetCtx, actor, actor->category, play);
     Target_InitReticle(targetCtx, actor->category, play);
 }
 
 void Target_Draw(TargetContext* targetCtx, PlayState* play) {
-    Actor* actor = targetCtx->lockOnActor;
+    Actor* actor; // used for both the reticle and arrow
+
+    actor = targetCtx->reticleActor;
```

Comment:
it seems like PRs recently have been lifting assignments up into the declaration when they can, why have you chosen to go the other way here and undo it?

Reply:
basically to raise importance to the fact that it is a multi use variable, and give room for my comment on that fact. I want to say "here is the first assignment to this variable" not "this is the only assignment to this variable"